### DISCUSSION TITLE
Date picker on data-table

### DISCUSSION THREAD

**Discussion by daryllvsantos** [shadcn-ui/ui#589] [Q&A]
Has anyone tried to add the Date-Picker as an option for filtering data on data-table

**Comment by R1NEJ0** [comment]
I'm trying to implement it

**Comment by R1NEJ0** [comment]
I've got it.

If you get the Taskt as example : https://github.com/shadcn-ui/ui/tree/main/apps/www/app/examples/tasks

In column definition: https://github.com/shadcn-ui/ui/blob/main/apps/www/app/examples/tasks/components/columns.tsx

Just add filterFn with it. 

Dont forget add column in toolbar component

Here I leave you and example of my code, I'm using date-fns to compare dates.

Toolbar Component
![image](https://github.com/shadcn-ui/ui/assets/26463620/c49bafd0-2ff5-4315-95cf-a7395e9927f3)

Column definition
![image](https://github.com/shadcn-ui/ui/assets/26463620/ce061442-1258-4908-beda-4e78684e5766)


Maybe it's not the best way to get it, but it is working.

**Comment by bpena707** [comment]
just ran into this because I want to impelemt something similar where I have the calendar at the top of the page maybe as a popup then select a date and filter the results on the data table also while having the default set to Today(). I have a data table set up with dates and custom time components I built but that is the only thing I am missing. do you have a live example I could look at?
